src/main/java/co/rsk/bitcoinj/script/Script.java

```diff
@@ -489,8 +490,51 @@ public Script getScriptSigWithSignature(Script scriptSig, byte[] sigBytes, int i
      * Returns the index where a signature by the key should be inserted. Only applicable to
      * a P2SH scriptSig.
      */
-    public int getSigInsertionIndex(Sha256Hash hash, BtcECKey signingKey) {
-        return this.redeemScriptParser.getSigInsertionIndex(hash, signingKey);
+    public int getSigInsertionIndex(Sha256Hash hashForSignature, BtcECKey signingKey) {
+        // Iterate over existing signatures, skipping the initial OP_0, the final redeem script
+        // and any placeholder OP_0 sigs.
+
+
+        // In order to affect consensus, we keep returning zero as the default value for the insertion index
+        // when the script is not a scriptSig.
+        final int defaultSigInsertionIndexForNoScriptSig = 0;
+        String noScriptSigFormatExceptionMessage = "Script does not match ScriptSig format";
+        if (chunks.size() < 2) {
```

Comment:
shouldn't we be checking for 4 instead of 2?
Since we are considering scriptSigs just for redeem scripts, according to [this documentation](https://learnmeabitcoin.com/technical/script/p2sh/) the scriptSig should have 4 chunks minimum

```
OP_0
sig1
sig2
redeemScriptChunk
```

Doing this change will help to discard non-script sigs faster :)
wdyt?

Reply:
Nice. I made the change, and the tests keep passing. 

However, are you sure that is possible to have a valid ScriptSig with just OP_0, redeemScriptChunk? @marcos-iov  @julia-zack

Reply:
it wouldn't make sense. Without signatures there's no way to unlock the script.

Reply:
if thats the case, the `getSigInsertionIndex` method couldn't return any valid index, right? 
And if you only have one signature you shouldn't be using a scriptSig but a scriptPubKey, as the doc says.

Reply:
You are argument looks pretty valid. I agree.